### ISSUE TITLE
Help Text Displays 'help' Option/Verb if AutoHelp is 'False'

### ISSUE THREAD

**Issue by x789** [commandlineparser/commandline#455]
### Description
If I configure the parser to not support the implicit option/verb 'help', the help text still contains the option/verb.
(This is possibly related to #414)

### Expected Behavior
I expect that the option/verb is not displayed.

### Environment
- .NET Core 2.2 Application
- CommandLineParser Nuget Package 2.5.0

### How to Reproduce
1. Configure the parser:
    ```
    var parser = new Parser(config =>
      {
        config.AutoHelp = false;
        config.HelpWriter = Console.Out;
      });
    var parserResult = parser.ParseArguments ...
    ```
2. Execute your program with `--help`
    > ERROR(S):
    >  Verb '--help' is not recognized.
    >
    >  --help       Display this help screen.
    >
    >  --version    Display version information.

**Comment by x789** [comment]
I investigated this issue and found a reason for this behavior. The parser-settings are not propagated to the method (`HelpText.AutoBuild`) that actually builds the help text.
Therefore, I think that #414 and #455 have the same origin.

I should be able to provide a patch in the next days...

**Comment by x789** [comment]
Closed [(Reason)](https://github.com/commandlineparser/commandline/pull/461#issuecomment-506968576)

**Comment by moh-hassan** [comment]
@x789 

The `AutoHelp /AutoVersion` properties of `Parser` for disabling AutoHelp /AutoVersion generation.
The `AutoHelp /AutoVersion` properties of `HelpText` for hiding/showing  --help/--version 
They have the same names but have different purpose.

So to hide --help/--version option , these properties should be false in both Parser and HelpText

var parser = new CommandLine.Parser(with=>
					{ 		
							
						with.AutoHelp=false; //disable AutoHelp Generation
						with.AutoVersion=false; //disable AutoVersion Generation						
					}); 

                              //use HelText to generate help without --help/--version options

                               var helpText = HelpText.AutoBuild(Result, h =>
							{					
								//this is working ONLY when parser  AutoHelp /AutoVersion =false
                // it may better if renamed ShowHelp/ShowVersion
								h.AutoHelp=false; //hide --help option
								h.AutoVersion=false;		//hide --version option									
								return h;
							}, 
							e =>e);

 
  It may be better if these properties are renamed in HelpText to be: ShowHelp/ShowVersion 

**How To show help/version** screen

In **WithNotParsed** method you should catch the error of type "**UnknownOptionError**" and based on the Token "help" or "version" show your custom help/version screen

**Comment by moh-hassan** [comment]
This issue is resolved in v2.6.0
Show [example](https://github.com/commandlineparser/commandline/issues/414#issuecomment-517824244)
